early-return/workflow.go

```diff
@@ -0,0 +1,79 @@
+package earlyreturn
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"go.temporal.io/sdk/workflow"
+)
+
+const (
+	UpdateName         = "early-return"
+	TaskQueueName      = "early-return-tq"
+	activityTimeout    = 2 * time.Second
+	earlyReturnTimeout = 5 * time.Second
```

Comment:
This is aggressive.  Our usual advice is to have more generous timeouts but to monitor latencies and keep them low.
Think about sitting at a computer waiting for an operation to complete.  If it was going to take more than 5 seconds, would you want it to fail saying "maybe that worked" ?  Or would you rather wait longer?  You'd probably be willing to wait a little longer.
Many client-facing RPC servers time out after around 30 seconds, and so these sorts of timeouts can be calibrated to be shorter than that.  
Here's a draft of a comment: 

One common heuristic: Calibrate this number relative to your overall remaining client timeout.  So, if your client will timeout in 29 more seconds, you might choose 28s to give time to return and report the correct error.
In general, err on the generous side so as not to fail operations that would have succeeded.

Reply:
Curious if @cretz 's advice would be similar.

Reply:
Completely situational I think. No strong opinion. Arguably the caller should determine how long they're willing to wait. A timer inside a workflow does not account for, say, the workflow being slow to start.

Reply:
- I've removed the "Await" timeout (`earlyReturnTimeout`) since it should be cancelled when the context times out. (see [other convo](https://github.com/temporalio/samples-go/pull/366#discussion_r1786974554))

- I've bumped the local activity timeout to 10s now; and the async activity to 30s.